aiocouch/couchdb.py

```diff
@@ -91,13 +91,13 @@ async def create(
 
         """
         db = Database(self, id)
-        if not await db._exists():
+        try:
             await db._put(**kwargs)
-            return db
-        elif exists_ok:
-            return db
-        else:
-            raise PreconditionFailedError(f"The database '{id}' does already exist.")
+        except PreconditionFailedError as e:
+            if not exists_ok:
```

Comment:
Is the only precondition that is checked here, that the database already exists? If not, we would bypass other precondition errors.

Reply:
According to the documentation, it is the only one: https://docs.couchdb.org/en/stable/api/database/common.html#put--db

While I agree that it would be better to check for the particular error, a string comparison of the error message isn't really future-proof. However, I cannot imagine another precondition upstream could add. If they decide to add new parameters, any issues with those would certainly raise a `BadRequest` instead. To be fair, I'd have used a 409 Conflict on CouchDB's side instead of the precondition, but it is what it is.

Reply:
Agree, the string check is not optimal, but your solution is completely fine for me.